### PR TITLE
Add ux-foundation, sass, postcss and image optimization to webpack

### PR DESCRIPTION
**Pull request by arzafran**
## Background
_UX folder structure was being imported from MSUXF but was copied in a weird way from node modules folder and it wasn't compliant with react's way of doing things, based in web components_
## Changes done
- Add sass-loader
- Add postcss-loader
- Add img-loder for image optimization
- Add ux folder structure for global styles
## Pending to be done
_Remove bootstrap, and add Component.scss to the templates that we generate via CLI_
## Notes
_N/A_

### CODE REVIEW

File: app/ux/styles.scss [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+/// - Main styles. Core content.
+///
+/// -> Components
+/// - Every modular/reusable element has its styles defined here.
+///
+/// -> Shame CSS
+/// - Yeah, we should be ashamed of this. MUST fix these in the near future.
+////
+
+@import "libs/**/*";
```
Comment (matiasbeckerle) [inline app/ux/styles.scss]: Was this import added on the other project (PR) you did this morning?

Comment (arzafran) [inline app/ux/styles.scss]: yeah. it was.

**Comment by AlphaGit** [comment]
Looks good -- I don't have much insight to provide into the framework itself.

However, should it be a bower / npm / NuGet package? Instead of copy-pasting the code we could import it and just update the references whenever updates were released. Is this something you guys are thinking about?

**Comment by arzafran** [comment]
@AlphaGit yeah, we have that working already but that's not the best solution, since we're adding only the folder structure, and not specific code (just core libs like jeet or normalize). The dependency way would be great if we're only importing an already compiled css, like what happens with bootstrap.

TL;DR: I don't feel comfortable enough with adding folder structures via npm, neither with copying and pasting it from node_modules folder (even though I've already created an npm task to automate the process)

I think, and this is more a pilosophical issue, the best approach is to have two frameworks, one for webapps and the other for websites. since having only one would fall short, or be an overkill, depending on which case it tries to solve.

Last, mantainance shouldn't be an issue since it's just folder structure, and webpack loaders.

**Comment by AlphaGit** [comment]
> The dependency way would be great if we're only importing an already compiled css, like what happens with bootstrap.

Why not with SASS code? (I don't disagree, I just didn't understand the reason.) I imagine it could be possible if the framework relied on some "hook" points (which is what every framework does), and if they could be optional, then even better, because it would be easy to integrate in every system that wants to use it.

**Comment by arzafran** [comment]
> Why not with SASS code? (I don't disagree, I just didn't understand the reason.) I imagine it could be possible if the framework relied on some "hook" points (which is what every framework does), and if they could be optional, then even better, because it would be easy to integrate in every system that wants to use it.

I'm not talking about the file extension, but with the code that bootstrap generates, which relies in you adding their markup and start coding right away, since we have a team of specialized UX devs, every style is going to be unique, and that's why we only provide a structure, and the basic wiring, with image optimization, postcss processing etc. instead of providing silver bullet styles. 

> I think I'm starting to see a point here. You expect this structure to change a lot on each individual project, right? So this framework is just some boilerplate starter code, rather than a framework. Is that correct?

Right. we don't intend to replace tools like bootstrap, instead we try to provide most of the tools needed to fasten up development as much as possible but still have the possibility to set creativity free and code any mindblowing design that UX designers provide to us.

:)

**Comment by AlphaGit** [comment]
@arzafran Gotcha. Thanks for the explanation. :)
